### ISSUE TITLE
Panic on Enabling PQ with Small Dataset

### ISSUE THREAD

**Issue by mohamedshahin-weaviate** [weaviate/weaviate#4381]
### How to reproduce this bug?

1. Start with a clean Weaviate instance.
2. Ingest a small dataset around 50 or 100 objects along with their vector representations.
3. Attempt to enable Product Quantization (PQ) on the ingested data

```
import traceback
import weaviate
import weaviate.collections.classes as wvc

def establish_connection():
    
    try:
        client = weaviate.connect_to_wcs(
            cluster_url="WCS_ENDPOINT",
            auth_credentials=weaviate.auth.AuthApiKey("AUTH_API_KEY"),
            headers={"X-OpenAI-Api-Key":"OPEN_AI_KEY"}
        )

        ready = client.is_ready()
        version = weaviate.__version__
        print(f"Weaviate client is ready: {ready}")
        print(f"Weaviate Version: {version}")
        return client

    except Exception as e:
        print(f"There is a problem: {e}")
        traceback.print_exc()
        return None
    
def enable_pq(client):
    d = client.collections.get("COLLECTION_NAME")
    d.config.update(
         vector_index_config=wvc.config.Reconfigure.VectorIndex.hnsw(
            quantizer=wvc.config.Reconfigure.VectorIndex.Quantizer.pq()
        )
    )
    
def get_logs(client):
    collection = client.collections.get("PDF")
    config = collection.config.get()
    pq_config = config.vector_index_config.pq
    # print some of the config properties
    print(f"Enabled: { pq_config.enabled }")
    print(f"Training: { pq_config.training_limit }")
    print(f"Segments: { pq_config.segments }")
    print(f"Centroids: { pq_config.centroids }")   

if __name__ == "__main__":
    client_connection = establish_connection()
    if client_connection:
        enable_pq(client_connection)
        get_logs(client_connection)
```

### What is the expected behavior?

Weaviate handle the scenario where the dataset is too small for PQ, without causing a panic crash.

### What is the actual behavior?

Weaviate crashes with a panic related to the PQ process when trying to enable PQ on a small dataset, leading to a restart.

```
panic: not enough data to fit kmeans

goroutine 5533695 [running]:
github.com/weaviate/weaviate/adapters/repos/db/vector/ssdhelpers.(*ProductQuantizer).Fit.func2(0xf0)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/ssdhelpers/product_quantization.go:365 +0x125
github.com/weaviate/weaviate/adapters/repos/db/vector/ssdhelpers.Concurrently.func1(0xf)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/ssdhelpers/utils.go:32 +0x7d
created by github.com/weaviate/weaviate/adapters/repos/db/vector/ssdhelpers.Concurrently in goroutine 5533670
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/ssdhelpers/utils.go:29 +0xe5
```

### Supporting information

- Weaviate Cloud Cluster (WCS) is used in the above example.
- Client: Python client (v4 release)
- Manually configure PQ (AutoPQ is not currently available in WCS)  
- The crash occurs specifically when enabling PQ on datasets of small size.
- Data size: Object count -> 62 & Dimensions stored -> 93696

### Server Version

1.23.10

### Code of Conduct

- [X] I have read and agree to the Weaviate's [Contributor Guide](https://weaviate.io/developers/contributor-guide) and [Code of Conduct](https://weaviate.io/service/code-of-conduct)

**Comment by rthiiyer82** [comment]
The fix for the above panic is available in 1.24.1 release. @mohamedshahin-weaviate : Please can you confirm that it is not seen any more?

**Comment by Shah91n** [comment]
@rthiiyer82 I still see panic on 1.24.1. 

```
panic: not enough data to fit kmeans

goroutine 5450912 [running]:
github.com/weaviate/weaviate/adapters/repos/db/vector/compressionhelpers.(*ProductQuantizer).Fit.func2(0xf0)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/compressionhelpers/product_quantization.go:392 +0x125
github.com/weaviate/weaviate/adapters/repos/db/vector/compressionhelpers.Concurrently.func1(0xf)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/compressionhelpers/utils.go:32 +0x7d
created by github.com/weaviate/weaviate/adapters/repos/db/vector/compressionhelpers.Concurrently in goroutine 5450826
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/compressionhelpers/utils.go:29 +0xe5
```

**Comment by rthiiyer82** [comment]
@mohamedshahin-weaviate : We have fixed this issue in 1.24.2 version. Please can you check at your end and confirm?

**Comment by Shah91n** [comment]
@rthiiyer82 It has no longer producing panic. Just error:

```
{
  "insertId": "zgn9gc3n80r8rg0x",
  "jsonPayload": {
    "level": "error",
    "msg": "Compressing vectors: not enough data to fit kmeans"
  },
```
